core/src/main/java/apoc/trigger/Trigger.java

```diff
@@ -18,41 +20,40 @@
-        public String name;
-        public String query;
-        public Map<String,Object> selector;
-        public Map<String, Object> params;
-        public boolean installed;
-        public boolean paused;
-
-        public TriggerInfo(String name, String query, Map<String, Object> selector, boolean installed, boolean paused) {
-            this.name = name;
-            this.query = query;
-            this.selector = selector;
-            this.installed = installed;
-            this.paused = paused;
-        }
+    // public for testing purpose
+    public static final String DB_NON_WRITER_ERROR = """
+            This instance is not allowed to write to the database because it is a follower.
+            Please use these procedures against the database leader.
+            """;
+    
+    @Context
+    public GraphDatabaseAPI db;
 
-        public TriggerInfo( String name, String query, Map<String,Object> selector, Map<String,Object> params, boolean installed, boolean paused )
-        {
-            this.name = name;
-            this.query = query;
-            this.selector = selector;
-            this.params = params;
-            this.installed = installed;
-            this.paused = paused;
-        }
-    }
+    @Context public TriggerHandler triggerHandler;
 
-    @Context public GraphDatabaseService db;
+    @Context public Log log;
 
-    @Context public TriggerHandler triggerHandler;
+    private void preprocessDeprecatedProcedures() {
+        final String msgDeprecation = """
+                Please note that the current procedure is deprecated,
+                it's recommended to use the `apoc.trigger.install`, `apoc.trigger.drop`, `apoc.trigger.dropAll`, `apoc.trigger.stop`, and `apoc.trigger.start` procedures,
+                instead of, respectively, `apoc.trigger.add`, `apoc.trigger.remove`, `apoc.trigger.removeAll`, `apoc.trigger.pause`, and `apoc.trigger.resume`.""";
+
+        log.warn(msgDeprecation);
 
-    @Procedure(name = "apoc.trigger.add", mode = Mode.WRITE)
+        if (!Util.isWriteableInstance(db)) {
+            throw new RuntimeException(DB_NON_WRITER_ERROR + msgDeprecation);
+        }
+    }
+
+    @Admin
+    @Deprecated
+    @Procedure(name = "apoc.trigger.add", mode = Mode.WRITE, deprecatedBy = "apoc.trigger.install")
     @Description("Adds a trigger to the given Cypher statement.\n" +
             "The selector for this procedure is {phase:'before/after/rollback/afterAsync'}.")
-    public Stream<TriggerInfo> add(@Name("name") String name, @Name("statement") String statement, @Name(value = "selector"/*, defaultValue = "{}"*/)  Map<String,Object> selector, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) {
+    public Stream<TriggerInfo> add(@Name("name") String name, @Name("kernelTransaction") String statement, @Name(value = "selector"/*, defaultValue = "{}"*/)  Map<String,Object> selector, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) {
```

Comment:
```suggestion
    public Stream<TriggerInfo> add(@Name("name") String name, @Name("statement") String statement, @Name(value = "selector")  Map<String,Object> selector, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) {
```

Reply:
The word "statement" is used in the docs, and in general I think it's a slightly clearer variable name than "kernelTransaction".